Classes/Eel/ElasticSearchQueryBuilder.php

```diff
@@ -596,13 +598,15 @@ public function fetch(): array
 
             $this->result['nodes'] = [];
 
-            $this->logThisQuery && $this->logger->debug(sprintf('Query Log (%s): %s -- execution time: %s ms -- Limit: %s -- Number of results returned: %s -- Total Results: %s', $this->logMessage, $request, (($timeAfterwards - $timeBefore) * 1000), $this->limit, count($searchResult->getHits()), $searchResult->getTotal()), LogEnvironment::fromMethodName(__METHOD__));
+            $this->logThisQuery && $this->logger->debug(sprintf('Query Log (%s): execution time: %s ms -- Limit: %s -- Number of results returned: %s -- Total Results: %s', $this->logMessage, (($timeAfterwards - $timeBefore) * 1000), $this->limit, count($searchResult->getHits()), $searchResult->getTotal()), LogEnvironment::fromMethodName(__METHOD__));
 
             if (count($searchResult->getHits()) > 0) {
                 $this->result['nodes'] = $this->convertHitsToNodes($searchResult->getHits());
             }
         } catch (ApiException $exception) {
-            $this->throwableStorage->logThrowable($exception);
+            $this->logThisQuery && $this->logger->debug(sprintf('Query Log (%s): execution failed with message: %s', $this->logMessage, $exception->getMessage()), LogEnvironment::fromMethodName(__METHOD__));
```

Comment:
Not sure, if we need two log messages here. The `logMessage` is helpful as often several queries are send when rendering a page. I would just prefix the error message with the `'Query Log (%s):` part and only send this one.

Reply:
I tweaked it a bit, but now I realize it's still not optimal.

- if `log()` was used when building the query, the query is logged before the exception log message
- without `log()` use, the exception is logged, but you don't know the query - hence I always log it now in case of an exception

This leads to exception + use of `log()` to have the query logged twice now - bad?